catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/DmsCoordinateProcessor.java

```diff
@@ -0,0 +1,219 @@
+    if (dmsSuggestion != null && dmsSuggestion.hasGeo()) {
+      LOGGER.trace("Adding the DMS suggestion to results [{}]", dmsSuggestion);
+      results.add(0, dmsSuggestion);
+    }
+    LOGGER.trace("(DMS) Done");
+  }
+
+  /**
+   * Deserialize a DMS string.
+   *
+   * @param dmsString the DMS string to deserialize.
+   * @return [dmsLat, dmsLon] The two deserialized parts of the DMS string.
+   */
+  private static DMSCoordinates parseDms(final String dmsString) {
+    final Matcher matcher = PATTERN_DMS_COORDINATES.matcher(dmsString);
+
+    if (matcher.matches()) {
+      final DMSComponent lat =
+          new DMSComponent(
+              matcher.group(DEGREES_LAT_GROUP),
+              matcher.group(MINUTES_LAT_GROUP),
+              matcher.group(SECONDS_LAT_GROUP),
+              matcher.group(DIRECTION_LAT_GROUP),
+              new DecimalFormat("00"));
+      final DMSComponent lon =
+          new DMSComponent(
+              matcher.group(DEGREES_LON_GROUP),
+              matcher.group(MINUTES_LON_GROUP),
+              matcher.group(SECONDS_LON_GROUP),
+              matcher.group(DIRECTION_LON_GROUP),
+              new DecimalFormat("000"));
+      return new DMSCoordinates(lat, lon);
+    }
+    return null;
+  }
+
+  private static String normalizedDmsString(final String dmsString) {
+
+    final DMSCoordinates dmsCoordinates = parseDms(dmsString);
+    final List<DMSComponent> dmsParts = Arrays.asList(dmsCoordinates.lat, dmsCoordinates.lon);
+    final NumberFormat minutesSecondsFormat = new DecimalFormat("00.###");
+
+    final StringBuilder dmsBuilder = new StringBuilder();
+    for (DMSComponent dmsPart : dmsParts) {
+      dmsBuilder
+          .append(dmsPart.degreeFormat.format(dmsPart.degrees))
+          .append("\u00B0")
+          .append(minutesSecondsFormat.format(dmsPart.minutes))
+          .append("\'")
+          .append(minutesSecondsFormat.format(dmsPart.seconds))
+          .append("\"")
+          .append(dmsPart.direction.toUpperCase())
+          .append(" ");
+    }
+    return dmsBuilder.toString().trim();
+  }
+
+  private static LatLon dmsToLatLon(final String dmsString) {
+    final DMSCoordinates dmsCoordinates = parseDms(dmsString);
+    if (dmsCoordinates == null) {
+      return null;
+    }
+
+    final DMSComponent dmsLat = dmsCoordinates.lat;
+    final DMSComponent dmsLon = dmsCoordinates.lon;
+    final int latModifier = dmsLat.direction.equalsIgnoreCase("N") ? 1 : -1;
+    final int lonModifier = dmsLon.direction.equalsIgnoreCase("E") ? 1 : -1;
+    final Double lat = toDecimalDegrees(dmsLat) * latModifier;
+    final Double lon = toDecimalDegrees(dmsLon) * lonModifier;
+
+    return (LatLon.isValidLatitude(lat) && LatLon.isValidLongitude(lon))
+        ? new LatLon(lat, lon)
+        : null;
+  }
+
+  private static Double toDecimalDegrees(final DMSComponent dms) {
+    return dms.degrees + dms.minutes / 60.0 + dms.seconds / 3600.0;
```

Comment:
❓ Could we move this function to `DMSComponent` itself? That would make for better encapsulation. This means it would take no input and no longer be static.

Reply:
good call